crates/matrix-sdk/src/room/mod.rs

```diff
@@ -368,6 +368,71 @@ impl Room {
         Ok(())
     }
 
+    /// Leave this room and all predecessors.
+    /// If any room was in [`RoomState::Invited`] state, it'll also be forgotten
+    /// automatically.
+    ///
+    /// Only invited and joined rooms can be left.
+    pub async fn leave(&self) -> Result<()> {
+        let mut rooms: Vec<Room> = vec![self.clone()];
+        let mut cur_room = self;
```

Comment:
`cur` isn't that shorter than `current`, let's avoid unnecessary abbreviations.

```suggestion
        let mut current_room = self;
```